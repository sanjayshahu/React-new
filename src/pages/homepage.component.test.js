import { shallow,mount,render } from "enzyme";
import { idText } from "typescript";
import {React}from "react";
//shallow:only main is rendered is js dom
//mount:main and its children are rendered in browser dom
//render:in hybrid platform of browser n jsdom
import HomePage from  './homepage.component'
import toJson from 'enzyme-to-json';

it('render',()=>{
    console.log(shallow(<HomePage/>))
})

it('renderlength',()=>{
    expect(shallow(<HomePage/>).length).toEqual(1);//only one component rendered because of shallow
})

it('take snapshot',()=>{
    expect( toJson(shallow(<HomePage/>))).toMatchSnapshot();//use this command for taking snapshot
}) 




