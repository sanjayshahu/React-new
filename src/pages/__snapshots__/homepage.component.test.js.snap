// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take snapshot 1`] = `
<div
  className="homepage"
>
  <h1>
    Welcome to  Homepage
  </h1>
  <Directory />
</div>
`;
